### ISSUE TITLE
Bugfix: Ocarina of Time - Forest Temple Poes

### ISSUE THREAD

**Issue by GoogleCodeExporter** [facelift/mupen64gc#171]
```
In the Forest Temple, when you are trying to join the parts of the puzzle
of the green Poo to invoke it, it just doesn't appear (I mean, the timer
doesn't even disappear). In other words, I can't continue. I've do this a
lot of times on some PC emulators without any problem. Is this a bug? I can
provide my save if you aren't that far in the game... I just wanna know I'm
not the only one that has this problem.
```

Original issue reported on code.google.com by `pabloacurielz@gmail.com` on 14 Nov 2009 at 6:01

**Comment by GoogleCodeExporter** [comment]
```
BTW, this happens on the v1.2 of the game (which I'm using). I have not tried 
with
the others ones...
```

Original comment by `pabloacurielz@gmail.com` on 14 Nov 2009 at 6:09

**Comment by GoogleCodeExporter** [comment]
```
... And no, this isn't a converted save...
```

Original comment by `pabloacurielz@gmail.com` on 14 Nov 2009 at 6:12

**Comment by GoogleCodeExporter** [comment]
```
I'm not sure, but a lot of the times, issues like this have something to do with
frame buffer textures or something. There's an option for that in the menu of 
the
emulator, if I recall correctly. Try playing around with that. And upload your 
save
anyway, dont wait for any of us or the devs to ask you for it.
```

Original comment by `DhrJo...@gmail.com` on 15 Nov 2009 at 2:44

**Comment by GoogleCodeExporter** [comment]
```
OK, I'll test that later. Here's my save. I hope someone can test that out, 
too...
```

Original comment by `pabloacurielz@gmail.com` on 15 Nov 2009 at 5:53

Attachments:
- [THE LEGEND OF ZELDA(U).sra](https://storage.googleapis.com/google-code-attachments/mupen64gc/issue-171/comment-4/THE LEGEND OF ZELDA%28U%29.sra)

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the save. This may be a core issue. 

What is happening and what is supposed to happen? Please be very detailed about 
it too.
```

Original comment by `emuki...@gmail.com` on 15 Nov 2009 at 11:18

**Comment by GoogleCodeExporter** [comment]
Original comment by `emuki...@gmail.com` on 15 Nov 2009 at 11:42
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
What's happening?
In the Forest Temple, one of the several dungeons of TLoZ:OoT, the player must 
find
and beat four Poos (ghosts), each one with a different color, to light the 
torches
found in the main room and thus be able to continue to the dungeon's boss 
(Phantom
Ganon). When you reach the third Poo room (a green one), you'll have to shoot an
arrow to the picture on the wall. Then, some puzzle parts will start to fall 
from the
ceiling (5, to be exact). You'll have to join 4 of them to recreate that 
picture.
When done correctly, the green Poo must appear, and it isn't, so I can't 
continue.

What's supposed to happen?
The green Poo must appear when the puzzle parts are joined.

If you haven't played Zelda, we'll need the help of someone else who has played 
the
game to reach that room.

P.S.: yeah, I know my english sucks...
```

Original comment by `pabloacurielz@gmail.com` on 16 Nov 2009 at 12:09

**Comment by GoogleCodeExporter** [comment]
```
... Thanks in advance, :D. It is a great job you've done with this emulator, 
guys.
```

Original comment by `pabloacurielz@gmail.com` on 16 Nov 2009 at 12:11

**Comment by GoogleCodeExporter** [comment]
```
pabloacurielz, I'm very familiar with Zelda OOT :) Is the issue that when you 
shoot
an arrow into the picture the Poe doesn't appear?

Please try this part in the pure interpreter core mode (change it from the 
settings,
then load your ROM and save). let me know if that works.
```

Original comment by `emuki...@gmail.com` on 16 Nov 2009 at 12:33

**Comment by GoogleCodeExporter** [comment]
```
IT DID WORK!!! Thanks emukidid, :D! So, it is core issue like you said some 
hours
ago, isn't it? Anyway, I can finally continue the game, haha...
```

Original comment by `pabloacurielz@gmail.com` on 16 Nov 2009 at 2:20

**Comment by GoogleCodeExporter** [comment]
```
Good to hear, yes it's a core issue.

Could you please provide instructions going from your save what I need to do to
replicate the issue? (It has been a while since I did the Forest Temple). Please
provide things like "Go to the left room, then right", etc.
```

Original comment by `emuki...@gmail.com` on 16 Nov 2009 at 2:45

**Comment by GoogleCodeExporter** [comment]
Original comment by `emuki...@gmail.com` on 16 Nov 2009 at 2:59
- Changed title: **Ocarina of Time - Forest Temple Poes**

**Comment by GoogleCodeExporter** [comment]
```
Hmmm... OK, here I go:

When you load my save, you'll appear at the Temple of Time (sorry about 
that...).
Just use the Ocarina of Time and play the Minuet of Forest to go to the Lost 
Woods.
You'll appear in front of the entrance to the Forest Temple (turn back). Use the
Hookshot to grab to the vine of the tree that is around there and enter the 
Temple.
Go straight until you get into the main room. Then, go to the left door. You'll 
be in
labyrinth-like room. Climb the two ladders and go to the left until you see a 
big
block. Get on it and go to the right until you see another block. Get on it and 
go
the right twice until you see another ladder. Climb it and go straight into the 
door.
Walk through the twisted corridor, then jump into the platforms to see another 
door.
Enter it and go down stairs until you see ANOTHER door (this will make you 
crazy).
Enter it and go through the door on the other side of the room. Go upstairs 
(this
room is an inverted replica of the another one) and go through the door. Now, 
jump
into the platforms to go to the corridor (pay atention to the hole on the 
wall). Go
straight and you'll see a door. Enter it. In this room, you'll have to melt the 
ice
in the switch to activate it (use Din's Fire in front of it [I know this isn't 
the
way to do it, but the fastest one]). Then, go back to the (now twisted) 
corridor. Go
to the platforms section on throw yourself into the hole I mentioned a while 
ago. In
this room, you'll have to avoid the falling roof and the Skulltulas (the big 
spiders)
and activate another switch (it's on the right side, search for it). Then, 
enter the
door and BINGO! You'll be on the puzzle room. Good luck...

LOL, that's a huge "hoopla". Remember that my english isn't very good. I'm 
doing my
best to make my explanations understandable.
```

Original comment by `pabloacurielz@gmail.com` on 16 Nov 2009 at 3:22

**Comment by GoogleCodeExporter** [comment]
```
... And, if you wonder why do I have the red tunic, I went to the Death 
Mountain only
once, to obtain it...
```

Original comment by `pabloacurielz@gmail.com` on 16 Nov 2009 at 3:26

**Comment by GoogleCodeExporter** [comment]
```
Thank you for the very detailed explanation. I'll try to figure this out 
tonight.
```

Original comment by `emuki...@gmail.com` on 16 Nov 2009 at 4:04

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Better yet, use my save state and SRAM. Btw, this is for the PAL "Master Quest"
edition rom; I don't know if the save data is compatible.

To reproduce:
Solve the puzzle by pushing the blocks together to form the image on the wall.  
One
of the blocks is superfluous and can be pushed aside.

Expected behavior:
A poe will appear, which you will have to fight in order to move on.

Observed behavior:
Poe does not appear, timer expires, and puzzle resets.
```

Original comment by `aan...@gmail.com` on 26 Nov 2009 at 8:29

Attachments:
- [THE LEGEND OF ZELDA(U).sra](https://storage.googleapis.com/google-code-attachments/mupen64gc/issue-171/comment-17/THE LEGEND OF ZELDA%28U%29.sra)
- [THE LEGEND OF ZELDA(U).st0](https://storage.googleapis.com/google-code-attachments/mupen64gc/issue-171/comment-17/THE LEGEND OF ZELDA%28U%29.st0)

**Comment by GoogleCodeExporter** [comment]
```
Odd that it's supposedly for the PAL version, and then you supply saves that are
appended with the USA delimiter (the "(U)" is what I mean by that)...
```

Original comment by `rhydonj@gmail.com` on 26 Nov 2009 at 11:48

**Comment by GoogleCodeExporter** [comment]
```
If that state doesn't work, then try mine:
```

Original comment by `rhydonj@gmail.com` on 27 Nov 2009 at 2:38

Attachments:
- [THE LEGEND OF ZELDA(U).st1](https://storage.googleapis.com/google-code-attachments/mupen64gc/issue-171/comment-19/THE LEGEND OF ZELDA%28U%29.st1)

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
Must be mislabeled, then; I was under the impression that Master Quest was 
PAL-only.
Also, another question: does anyone have trouble with the pots in the puzzle 
block
room?  Playing through again, I've noticed that the emulator freezes whenever I 
try
to break one of the pots, either by throwing it or slicing it.  This does not 
happen
in Pure Interpreter mode.  If anyone can replicate, I will open another bug 
report.

```

Original comment by `aan...@gmail.com` on 27 Nov 2009 at 5:37

**Comment by GoogleCodeExporter** [comment]
```
@aanel: I had no problem with them. Once I defeated the Green Poe, I saved the 
game
and I went back to the Wii64 menu, to re-activate the Dynarec mode. BTW, thanks 
for
those observations, :).
```

Original comment by `pabloacurielz@gmail.com` on 27 Nov 2009 at 8:26

**Comment by GoogleCodeExporter** [comment]
```
emukidid. honor who deservs honor you are great, I expend 1 hour for nothing 
try to 
resolve the puzzle in forest temple, I have the 64 version, collector´s 
edition and 
virtual console of this wonderful game and you can see I played this game a lot 
of 
times and never though that this was a issue core thanks for the cheat in 
setting,Im 
from mexico thats why my english is not very good i hope you understand,y como 
decimos en mexico "Honor a quien Honor merece, eres chingon" excelente New 
year!!!
```

Original comment by `samibiz...@hotmail.com` on 3 Jan 2010 at 6:32

**Comment by GoogleCodeExporter** [comment]
Original comment by `emuki...@gmail.com` on 1 Feb 2010 at 11:33
- Changed title: **Bugfix: Ocarina of Time - Forest Temple Poes**

**Comment by GoogleCodeExporter** [comment]
```
I am having the exact same problem. Could you explain how you fixed it?
```

Original comment by `danny.sa...@gmail.com` on 22 May 2010 at 12:34

**Comment by GoogleCodeExporter** [comment]
```
I thought it was fixed with the Beta 1.1... If it wasn't, you have to go to the
Options menu on the main screen and change the core mode from "Dynarec" to
"Interpreter". Then, play the game just as you have been doing (the game will 
run
very SLOW). When you beat that part, you can readjust the settings so you can 
use the
Dynarec mode. Good luck,;).
```

Original comment by `pabloacurielz@gmail.com` on 22 May 2010 at 3:47

**Comment by GoogleCodeExporter** [comment]
```
Poes, not Poos.

I don't want the Wii64 team getting involved with any Green Poos.
```

Original comment by `school.p...@gmail.com` on 13 Sep 2011 at 1:44

**Comment by GoogleCodeExporter** [comment]
```
Sorry, engrish, can't help it. It should be noted that my mother tongue is 
Spanish, but I'm trying my best to improve my English.

Good joke, btw.
```

Original comment by `pabloacurielz@gmail.com` on 13 Sep 2011 at 6:48

**Comment by GoogleCodeExporter** [comment]
```
Confirmed fixed in > v1.1
```

Original comment by `emuki...@gmail.com` on 9 Oct 2012 at 11:40
- Changed state: **Fixed**
